tests/python/gaia-ui-tests/gaiatest/apps/system/regions/activities.py

```diff
@@ -29,36 +29,48 @@ def __init__(self, marionette):
         Wait(self.marionette).until(lambda m: view.location['y'] == 0)
 
     def tap_wallpaper(self):
+        Wait(self.marionette).until(
+            expected.element_displayed(*self._actions_menu_locator))
         self.marionette.find_element(*self._wallpaper_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        Wait(self.marionette).until(
+            expected.element_not_displayed(*self._actions_menu_locator))
         from gaiatest.apps.wallpaper.app import Wallpaper
         wallpaper = Wallpaper(self.marionette)
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == wallpaper.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == wallpaper.name)
         self.apps.switch_to_displayed_app()
         return wallpaper
 
     def tap_gallery(self):
         self.marionette.find_element(*self._gallery_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
+        Wait(self.marionette).until(
+            expected.element_not_displayed(actions_menu))
         from gaiatest.apps.gallery.app import Gallery
         gallery = Gallery(self.marionette)
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == gallery.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == gallery.name)
         self.apps.switch_to_displayed_app()
         return gallery
 
     def tap_camera(self):
         self.marionette.find_element(*self._camera_button_locator).tap()
-        self.wait_for_element_not_displayed(*self._actions_menu_locator)
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
+        Wait(self.marionette).until(
+            expected.element_not_displayed(actions_menu))
         from gaiatest.apps.camera.app import Camera
         camera = Camera(self.marionette)
-        self.wait_for_condition(lambda m: self.apps.displayed_app.name == camera.name)
+        Wait(self.marionette).until(lambda m: self.apps.displayed_app.name == camera.name)
         self.apps.switch_to_displayed_app()
         camera.wait_for_capture_ready()
         return camera
 
     def tap_cancel(self):
+        actions_menu = Wait(self.marionette).until(
+            expected.element_present(*self._actions_menu_locator))
```

Comment:
I'd prefer if we waited for the element to be visible, as we're explicitly waiting based on visibility after the tap.